contrib/checkstyle/src/mill/contrib/checkstyle/CheckstyleModule.scala

```diff
@@ -0,0 +1,195 @@
+package mill
+package contrib.checkstyle
+
+import mill.api.{Loose, PathRef}
+import mill.scalalib.{DepSyntax, JavaModule}
+import mill.util.Jvm
+
+import javax.xml.transform.TransformerFactory
+import javax.xml.transform.stream.{StreamResult, StreamSource}
+
+/**
+ * Performs quality checks on Java source files using [[https://checkstyle.org/ Checkstyle]] and generates reports from these checks.
+ */
+trait CheckstyleModule extends JavaModule {
+
+  import CheckstyleModule._
+
+  /**
+   * Generates a [[Checkstyle]].
+   */
+  def checkstyle: T[Checkstyle] = T {
+
+    val dest = T.dest
+
+    val format = checkstyleFormat()
+    val report = dest / s"checkstyle-report.$format"
+
+    val args = checkstyleOptions() ++
+      Seq(
+        "-c",
+        checkstyleConfig().path.toString(),
+        "-f",
+        format,
+        "-o",
+        report.toString()
+      ) ++
+      sources().map(_.path.toString())
+
+    T.log.info("generating checkstyle report ...")
+    T.log.debug(s"running checkstyle with $args")
+
+    val errors = Jvm.callSubprocess(
+      mainClass = "com.puppycrawl.tools.checkstyle.Main",
+      classPath = checkstyleClasspath().map(_.path),
+      mainArgs = args,
+      workingDir = dest,
+      check = false
+    ).exitCode
```

Comment:
This is a bit of an odd API having the exit code be the number of errors, given it's what's available upstream that's fine. But what happens if there are more than 255 errors in detected? Does it cap out, or wrap around, or something else? Whichever behavior it is should document it

Reply:
I think somewhere along the way the `byte` value for -1 is being transformed into an unsigned `int`.
Will update the comment.